### ISSUE TITLE
CUDA errors are not handled properly: conversion from bytes to str required

### ISSUE THREAD

**Issue by DenisKuplyakov** [onnx/onnx-tensorrt#728]
## Description

It looks like that error strings from cuda functions should be converted from bytes to str before using them.

```
>>> engine = backend.prepare(m, device="CUDA:1")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/onnx_tensorrt/backend.py", line 248, in prepare
    return TensorRTBackendRep(model, device, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/onnx_tensorrt/backend.py", line 56, in __init__
    self._set_device(device)
  File "/usr/local/lib/python3.8/dist-packages/onnx_tensorrt/backend.py", line 156, in _set_device
    cudaSetDevice(device.device_id)
  File "/usr/local/lib/python3.8/dist-packages/onnx_tensorrt/backend.py", line 39, in cudaSetDevice
    raise RuntimeError("cudaSetDevice: " + error_string)
TypeError: can only concatenate str (not "bytes") to str
```

Now the problem masks real error messages.

## Environment

**TensorRT Version**: 
**ONNX-TensorRT Version / Branch**: 21.05
**GPU Type**: 1080 TI
**Nvidia Driver Version**: 465.19.01
**CUDA Version**:  11.3.0
**CUDNN Version**: 8.2.0.51
**Operating System + Version**: Ubuntu 20.04
**Python Version (if applicable)**: 3.8.5
**TensorFlow + TF2ONNX Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**:  nvcr.io/nvidia/tensorrt:21.05-py3

## Steps To Reproduce
1. Pass `device=CUDA:100` to `backend.prepare()`

**Comment by jackwish** [comment]
@DenisKuplyakov Thanks for reporting this issue. It seems to be an issue of the data type handling between ctypes and python. Will try to create a fix.

**Comment by jackwish** [comment]
This has been fixed by https://github.com/onnx/onnx-tensorrt/pull/859

@DenisKuplyakov  Could you please try if issue resolved at your side?

**Comment by jackwish** [comment]
Closing as the fix has been merged. Feel free to reopen if any issue still. Thanks!

**Comment by DenisKuplyakov** [comment]
> This has been fixed by #859
> 
> @DenisKuplyakov Could you please try if issue resolved at your side?

Thanks a lot! But we have switched from onnx-tensorrt to onnxruntime with tensorrt backend in every project. So I have no ability to verify the fix. Sorry
